BepInEx.EnableResize/EnableResize.cs

```diff
@@ -62,24 +76,43 @@ internal void Awake()
             }, IntPtr.Zero);
 
             if (WindowHandle == IntPtr.Zero) return;
-            SceneManager.sceneLoaded += (s, lsm) =>
-            {
-                if (lsm == LoadSceneMode.Single) ResizeWindow();
-            };
+
+            StartCoroutine(TestScreen());
+            ConfigEnableResize.SettingChanged += (sender, args) =>  StartCoroutine(TestScreen());
         }
 
-        internal void Update()
+        private IEnumerator TestScreen()
         {
-            if (!ConfigEnableResize.Value) return;
-            var fs = Screen.fullScreen;
-            if (!fs && prev) ResizeWindow();
-            prev = fs;
+            while (true)
+            {
+                if (!ConfigEnableResize.Value) yield break;
+
+                fs = Screen.fullScreen;
+                res = Screen.width + Screen.height;
+                style = GetWindowLong(WindowHandle, GWL_STYLE);
+
+                // If zero, is in borderless mode
+                borderless = style & borderlessMask;
+
+                if (!fs && prevFS ||
+                    res != prevRes ||
+                    borderless != 0 && prevBorderless == 0)
+                {
+                    ResizeWindow();
+                }
+
+                prevBorderless = borderless;
+                prevFS = fs;
+                prevRes = res;
+                yield return oneSecond;
```

Comment:
Could this be simplified somewhat? Maybe more descriptive field names will help make it more readable.

Reply:
Couldn't simplify, but the new fields seems to be helping